presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoPreparedStatement.java

```diff
@@ -858,4 +876,69 @@ private static String typedNull(String prestoType)
+            throws SQLException
+    {
+        ImmutableList.Builder<ColumnInfo> list = ImmutableList.builder();
+        while (resultSet.next()) {
+            String columnName = resultSet.getString("Column Name");
+            String catalog = resultSet.getString("Catalog");
+            String schema = resultSet.getString("Schema");
+            String table = resultSet.getString("Table");
+            ClientTypeSignature clientTypeSignature = getClientTypeSignatureFromTypeString(resultSet.getString("Type"));
+            ColumnInfo.Builder builder = new ColumnInfo.Builder()
+                    .setColumnName(columnName)
+                    .setColumnLabel(columnName)
+                    .setCatalogName(catalog)
+                    .setSchemaName(schema)
+                    .setTableName(table)
+                    .setColumnTypeSignature(clientTypeSignature)
+                    .setNullable(ColumnInfo.Nullable.UNKNOWN);
+            setTypeInfo(builder, clientTypeSignature);
+            list.add(builder.build());
+        }
+        return list.build();
+    }
+
+    @VisibleForTesting
+    static ClientTypeSignature getClientTypeSignatureFromTypeString(String type)
+    {
+        String topLevelType;
+        List<ClientTypeSignatureParameter> arguments = new ArrayList<>();
+        Matcher topLevelMatcher = TOP_LEVEL_TYPE_PATTERN.matcher(type);
+        if (topLevelMatcher.matches()) {
+            topLevelType = topLevelMatcher.group(1);
+            String typeParameters = topLevelMatcher.group(2);
+            if (topLevelType.equals("decimal")) {
+                List<String> precisionAndScale = Splitter.on(',').splitToList(typeParameters);
+                checkArgument(precisionAndScale.size() == 2, "Invalid decimal parameters: %s", typeParameters);
+                arguments.add(ClientTypeSignatureParameter.ofLong(parseLong(precisionAndScale.get(0))));
+                arguments.add(ClientTypeSignatureParameter.ofLong(parseLong(precisionAndScale.get(1))));
+            }
+            else if (topLevelType.equals("char") || topLevelType.equals("varchar")) {
+                long precision = parseLong(typeParameters);
+                arguments.add(ClientTypeSignatureParameter.ofLong(precision));
+            }
+        }
+        else {
+            Matcher timestampMatcher = TIMESTAMP_WITH_TIME_ZONE_PRECISION_PATTERN.matcher(type);
+            Matcher timeMatcher = TIME_WITH_TIME_ZONE_PRECISION_PATTERN.matcher(type);
+            if (timestampMatcher.matches()) {
+                topLevelType = "timestamp with time zone";
+                arguments.add(ClientTypeSignatureParameter.ofLong(parseLong(timestampMatcher.group(1))));
+            }
+            else if (timeMatcher.matches()) {
+                topLevelType = "time with time zone";
+                arguments.add(ClientTypeSignatureParameter.ofLong(parseLong(timeMatcher.group(1))));
+            }
+            else {
+                topLevelType = type;
```

Comment:
@electrum should we have 
```
verify(!type.contains('('), "Unhandled parameterized type: %s", type);
```

here?

Reply:
This code is only used for metadata display purposes, so I don't think we want to fail here. Seems fine to ignore anything that is not supported.